src/Symfony/Component/Validator/Constraints/NoSuspiciousCharacters.php

```diff
@@ -0,0 +1,128 @@
+{
+    public const RESTRICTION_LEVEL_ERROR = '1ece07dc-dca2-45f1-ba47-8d7dc3a12774';
+    public const INVISIBLE_ERROR = '6ed60e6c-179b-4e93-8a6c-667d85c6de5e';
+    public const CHAR_LIMIT_ERROR = 'ae6e496e-b315-4fdf-bea5-657accac631d';
+    public const MIXED_NUMBERS_ERROR = '9f01fc26-3bc4-44b1-a6b1-c08e2412053a';
+    public const HIDDEN_OVERLAY_ERROR = '56380dc5-0476-4f04-bbaa-b68cd1c2d974';
+
+    protected const ERROR_NAMES = [
+        self::RESTRICTION_LEVEL_ERROR => 'RESTRICTION_LEVEL_ERROR',
+        self::INVISIBLE_ERROR => 'INVISIBLE_ERROR',
+        self::CHAR_LIMIT_ERROR => 'CHAR_LIMIT_ERROR',
+        self::MIXED_NUMBERS_ERROR => 'MIXED_NUMBERS_ERROR',
+        self::HIDDEN_OVERLAY_ERROR => 'INVALID_CASE_ERROR',
+    ];
+
+    /**
+     * Check that a string satisfies the requirements for the specified restriction level.
+     * It defaults to {@see self::RESTRICTION_LEVEL_HIGH} when using ICU >= 58,
+     * and is locked to {@see self::RESTRICTION_LEVEL_SINGLE_SCRIPT} on older versions.
+     */
+    public const CHECK_RESTRICTION_LEVEL = 16;
+
+    /**
+     * Check a string for the presence of invisible characters such as zero-width spaces,
+     * or character sequences that are likely not to display such as multiple occurrences of the same non-spacing mark.
+     */
+    public const CHECK_INVISIBLE = 32;
+
+    /** Check a string contains only characters allowed by the configured profile. */
+    public const CHECK_CHAR_LIMIT = 64;
+
+    /**
+     * Check that a string does not mix numbers from different numbering systems;
+     * for example “8” (Digit Eight) and “৪” (Bengali Digit Four).
+     */
+    public const CHECK_MIXED_NUMBERS = 128;
+
+    /**
+     * Check that a string does not have a combining character following a character in which it would be hidden;
+     * for example “i” (Latin Small Letter I) followed by a U+0307 (Combining Dot Above).
+     */
+    public const CHECK_HIDDEN_OVERLAY = 256;
+
+    public const RESTRICTION_LEVEL_ASCII = 268435456;
+    public const RESTRICTION_LEVEL_SINGLE_SCRIPT = 536870912;
+    public const RESTRICTION_LEVEL_HIGH = 805306368;
+    public const RESTRICTION_LEVEL_MODERATE = 1073741824;
+    public const RESTRICTION_LEVEL_MINIMAL = 1342177280;
+    public const RESTRICTION_LEVEL_NONE = 1610612736;
+
+    public string $restrictionLevelMessage = 'Restriction level check failed.';
+    public string $invisibleMessage = 'Invisible check failed.';
+    public string $charLimitMessage = 'Char limit check failed.';
+    public string $mixedNumbersMessage = 'Mixed numbers check failed.';
+    public string $hiddenOverlayMessage = 'Hidden overlay check failed.';
+
+    public int $checks = self::CHECK_RESTRICTION_LEVEL | self::CHECK_INVISIBLE | self::CHECK_CHAR_LIMIT | self::CHECK_MIXED_NUMBERS | self::CHECK_HIDDEN_OVERLAY;
+    public ?int $restrictionLevel = null;
+    public array $profileLocales = [];
+    public bool $addCurrentLocaleToProfile = true;
```

Comment:
I’m not sure about this default value :thinking:

Reply:
ah, thanks for pointing this out
I'm not sure this should be `LocaleAwareInterface` indeed.
Can't we set this to the list to the `enabled_locales` of the app instead when the attribute doesn't specify them?

Reply:
Well `LocaleAwareInterface` would allow internationalized apps to check inputs relative to the user’s locale so I thought it was necessary, but I’m not sure this should be enabled by default.

If it is disabled we could indeed default `$profileLocales` to `enabled_locales`.

Reply:
Why would an app that enabled EN and GR only accept one of those when displaying in EN?
To me it should accept both.

Reply:
I would say it’s up to the app to decide if Greek characters submitted by an english user are suspicious.

Reply:
Then ppl will write their own constraint to me. I don't see why we should support this complexity.

Reply:
Okay this is a way to answer the question :grinning: 

So I’ll remove `$addCurrentLocaleToProfile` and make `$profileLocales` defaults to `enabled_locales`.